sdks/common/runner-api/src/main/proto/beam_runner_api.proto

```diff
@@ -648,34 +612,44 @@ message Environment {
   string url = 1;
 }
 
-// Description of a function in a Beam pipeline.
+// A specification of a user defined function.
 //
-// Contains one of _or both of_ a UrnWithParameter specifying the function
-// and the specification for how to execute it against a particular
-// SDK's harness.
 message FunctionSpec {
 
-  // (Optional) An SDK-independent specification of this function.
-  // If present, this must _fully_ specify the function.
+  // (Required) A full specification of this function.
+  //
+  // There are two main styles, which need not be fully distinct
   //
-  // For example the distinguished urn "urn:beam:windowfn:FixedWindows" with
-  // payload `{ duration: n }` fully specifies a windowing function which can
-  // be implemented by the SDK constructing the pipeline, by another SDK (for
-  // language-to-language fusion compatibility) or by the runner directly.
+  // 1. A well-known URN like "urn:beam:windowfn:FixedWindows"
```

Comment:
Done. Side note: I think later we might rename `UrnWithParameter` to `FunctionSpec` and `FunctionSpec` to `FunctionSpecAndEnvironment` or some such. I dislike `UrnWithParameter` but I'm trying not to rename it in this PR.